src/flows/shared/steps/OrganizeCollection/Sidebar/Sidebar.tsx

```diff
@@ -2,231 +2,179 @@ import { memo, useCallback, useMemo, useState } from 'react';
+  const isPOAPEnabled = isFeatureEnabled(FeatureFlag.POAP, query);
 
-  const tokens = removeNullValues(allTokens);
-
-  const [debouncedSearchQuery, setDebouncedSearchQuery] = useState('');
+  const { stageTokens } = useCollectionEditorActions();
   const [searchResults, setSearchResults] = useState<string[]>([]);
+  const [selectedChain, setSelectedChain] = useState<Chain>('Ethereum');
+  const [debouncedSearchQuery, setDebouncedSearchQuery] = useState('');
+
+  const isSearching = debouncedSearchQuery.length > 0;
 
+  const nonNullTokens = removeNullValues(allTokens);
   const sidebarTokensAsArray = useMemo(() => convertObjectToArray(sidebarTokens), [sidebarTokens]);
 
-  const tokensFilteredBySearch = useMemo(() => {
-    if (debouncedSearchQuery) {
-      const searchResultNfts = [];
-      for (const resultId of searchResults) {
-        if (sidebarTokens[resultId]) {
-          searchResultNfts.push(sidebarTokens[resultId]);
-        }
-      }
+  const editModeTokensSearchResults = useMemo(() => {
+    if (!debouncedSearchQuery) {
+      return sidebarTokensAsArray;
+    }
 
-      return searchResultNfts;
+    const searchResultNfts = [];
+    for (const resultId of searchResults) {
+      if (sidebarTokens[resultId]) {
+        searchResultNfts.push(sidebarTokens[resultId]);
+      }
     }
 
-    return sidebarTokensAsArray;
+    return searchResultNfts;
   }, [debouncedSearchQuery, searchResults, sidebarTokens, sidebarTokensAsArray]);
 
-  const nftFragmentsKeyedByID = useMemo(() => keyBy(tokens, (token) => token.dbid), [tokens]);
+  const nftFragmentsKeyedByID = useMemo(
+    () => keyBy(nonNullTokens, (token) => token.dbid),
+    [nonNullTokens]
+  );
 
-  const nonNullTokens = useMemo(() => {
-    return tokensFilteredBySearch.filter((editModeToken) =>
-      Boolean(nftFragmentsKeyedByID[editModeToken.id])
-    );
-  }, [nftFragmentsKeyedByID, tokensFilteredBySearch]);
+  const handleAddBlankBlockClick = useCallback(() => {
+    const id = `blank-${generate12DigitId()}`;
+    stageTokens([{ id, whitespace: 'whitespace' }]);
+    // auto scroll so that the new block is visible. 100ms timeout to account for async nature of staging tokens
+    setTimeout(() => {
+      document.getElementById(id)?.scrollIntoView({ behavior: 'smooth' });
+    }, 100);
+  }, [stageTokens]);
+
+  const tokensToDisplay = useMemo(() => {
+    return editModeTokensSearchResults.filter((editModeToken) => {
+      const token = nftFragmentsKeyedByID[editModeToken.id];
+
+      // Ensure we have a 1-1 match.
+      // Every EditModeToken should have a Token from Relay
+      if (!token) {
+        return false;
+      }
 
-  const { isRefreshingNfts, handleRefreshNfts } = useWizardState();
+      // If we're searching, we want to search across all chains
+      if (isSearching) {
+        // Unless POAP is disabled, then we want ONLY ETH STUFF
+        if (!isPOAPEnabled) {
+          return token.chain === 'Ethereum';
```

Comment:
could be useful to leave a note here to indicate this'll need to be updated to support tezos? 
totally optional lol, might make it easier to remember next week 🤷‍♂️